File: Solutions/779. K-th Symbol in Grammar/779.java
class Solution {
    public int kthGrammar(int n, int k) {
        boolean isinverted = false;
        n = (int) Math.pow(2, n-1);
        while(n != 1){
            n /= 2;
            if (k > n){
                k -= n;
                isinverted = !isinverted;
            }
        }
        return isinverted ? 1 : 0;
    }
}

File: Solutions/779. K-th Symbol in Grammar/779.py
class Solution:
    def kthGrammar(self, n: int, k: int) -> int:
        isinverted = False
        n = 2**(n-1)
        while n != 1:
            n /= 2
            if k>n :
                k -= n
                isinverted = not isinverted
        return 1 if isinverted else 0